R/data.table.R

```diff
@@ -49,6 +49,32 @@ null.data.table = function() {
   setalloccol(ans)
 }
 
+rowwiseDT = function(..., key=NULL) {
+  x = substitute(list(...))[-1L]
+  if (is.null(nms <- names(x)))
+    stop("Must provide at least one column (use `name=`)")
+  header_pos = which(nzchar(nms))
+  if (any(nzchar(x[header_pos])))
+    stop("Named arguments must be empty")
+  if (!identical(header_pos, seq_along(header_pos)))
+    stop("Header must be the first N arguments")
+  header = nms[header_pos]
+  ncols = length(header)
+  body = lapply(x[-header_pos], eval, envir = parent.frame())
```

Comment:
Is it common to include default arguments? 
```
eval(expr, envir = parent.frame(),
           enclos = if(is.list(envir) || is.pairlist(envir))
                       parent.frame() else baseenv())
```

Reply:
Sorry but I don't understand why eval at parent.frame is not good.

Reply:
It is actually not obvious if a default argument is evaluated having same parent.frame() as if we provide that during function call.

Reply:
@ColeMiller1 I just hit by this kind of usage. `eval(expr)` returned a different answer than `eval(expr, parent.frame())`.